supervisor/api/__init__.py

```diff
@@ -751,7 +751,10 @@ def _register_store(self) -> None:
         )
 
     def _register_panel(self) -> None:
-        """Register panel for Home Assistant."""
+        """Register panel for Home Assistant.
+
+        Must be run in executor.
```

Comment:
Why? `joinpath` should be fine no? :thinking:

Reply:
Hm, I see, surprising, but ok. It would mean we need to call from a executor. Can we consider aiohttp a thread safe library? :thinking:

Reply:
Ok after discussion I think I figured out how to separate making the static asset and modifying the webapp object so we aren't modifying the common object in the executor. I also added a testcase for frontend assets so I can confirm it works before and after my changes here.